terraform/modules/benchmark/service/bench.rb

```diff
@@ -0,0 +1,343 @@
+end
+
+# Convert raw seconds to a nicer format for readability, displaying one decimal
+# point if less than `60` seconds and rounding to whole numbers if greater than
+# `60` seconds.
+def trunc_seconds(seconds)
+  seconds < MINUTE ? seconds.ceil(1) : seconds.ceil(0)
+end
+
+# Updates physical and virutal memory size metrics.
+def sample_proc(pid, metrics)
+  output = syscall('ps', '-o', 'rss,vsz', pid.to_s)
+  output = output.split("\n").last.split
+  metrics[:rss].push(output[0].to_i)
+  metrics[:vsz].push(output[1].to_i)
+end
+
+# Populate import table with metrics.
+def format_import_table_row(key, value)
+  elapsed = value[:import][:elapsed] || 'n/a'
+  rss = value[:import][:rss]
+  rss_max = rss ? (rss.max * 1024).to_bibyte : 'n/a'
+  vsz = value[:import][:vsz]
+  vsz_max = vsz ? (vsz.max * 1024).to_bibyte : 'n/a'
+  db_size = value[:import][:db_size] || 'n/a'
+  "#{key} | #{elapsed} | #{rss_max} | #{vsz_max} | #{db_size}\n"
+end
+
+# Format import table and call function to populate metrics.
+def format_import_table(metrics)
+  result = "Bench | Import Time [sec] | Import RSS | Import VSZ | DB Size\n"
+  result += "-|-|-|-|-\n"
+
+  metrics.each do |key, value|
+    result += format_import_table_row(key, value)
+  end
+
+  result
+end
+
+# Format validation table and populate table with metric.
+def format_validate_table(metrics)
+  result = "Bench | Validate Time [sec] | Validate RSS | Validate VSZ\n"
+  result += "-|-|-|-\n"
+
+  metrics.each do |key, value|
+    elapsed = value[:validate][:elapsed] || 'n/a'
+    rss = value[:validate][:rss]
+    rss_max = rss ? (rss.max * 1024).to_bibyte : 'n/a'
+    vsz = value[:validate][:vsz]
+    vsz_max = vsz ? (vsz.max * 1024).to_bibyte : 'n/a'
+    result += "#{key} | #{elapsed} | #{rss_max} | #{vsz_max}\n"
+  end
+
+  result
+end
+
+# Output database benchmark metrics to markdown file.
+def write_markdown(metrics)
+  # Output file is a suite of markdown tables.
+  result = ''
+  result += format_import_table(metrics)
+  result += "---\n"
+  result += format_validate_table(metrics)
+
+  filename = "result_#{Time.now.to_i}.md"
+  File.write(filename, result)
+  @logger.info "Wrote #{filename}"
+end
+
+# Output daily benchmark metrics to comma-separated value file.
+def write_csv(metrics, options)
+  filename = "result_#{Time.now.to_i}.csv"
+  CSV.open(filename, 'w') do |csv|
+    csv << ['Timestamp', 'Forest Version', 'Lotus Version', 'Chain', 'Metric', 'Forest Value', 'Lotus Value']
+
+    timestamp = Time.now.strftime('%Y-%m-%d %H:%M:%S')
+    chain = options[:chain]
+
+    results = { import_time: { forest: 'n/a', lotus: 'n/a' },
+                validation_time: { forest: 'n/a', lotus: 'n/a' } }
+
+    metrics.each do |key, value|
+      elapsed = value[:import][:elapsed] || 'n/a'
+      tpm = value[:validate_online][:tpm] || 'n/a'
+
+      results[:import_time][key.to_sym] = "#{elapsed} sec"
+      results[:validation_time][key.to_sym] = "#{tpm} tipsets/sec"
```

Comment:
It's wasteful to include the unit in every row of the result. This should be a part of the header.

Reply:
`import_time` and `validation_time` comes under `Metric` header, each metric will have its own unit
we will have to restructure the output format 
I can take this out in a separate PR